### PR TITLE
Remove Accept/Decline from Alice's actions

### LINKED ISSUE #561: Alice should not have Accept/Decline actions
Make the enum that Alice returns from her `actions` method not have `Accept` & `Decline` variants.
Alice & Bob should have a different enum.

https://github.com/coblox/comit-rs/blob/10f6179bd66c95f0d785d397536160782acc651d/application/comit_node/src/http_api/rfc003/action.rs#L112 Should be removed as a consequence.

Blocks #411 
Blocks #576

### PR DESCRIPTION
**Pull request by luckysori**
Resolves #561.

### CODE REVIEW

File: application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_erc20.rs [hunk 1]
```diff
@@ -49,24 +49,28 @@ impl OngoingSwap<Alice<Bitcoin, Ethereum, BitcoinQuantity, Erc20Quantity>> {
...
+            }
             SS::BothFunded(BothFunded {
                 ref alpha_htlc_location,
                 ref beta_htlc_location,
                 ref swap,
                 ..
             }) => vec![
-                ActionKind::Redeem(swap.redeem_action(*beta_htlc_location)),
-                ActionKind::Refund(swap.refund_action(*alpha_htlc_location)),
+                alice::ActionKind::Redeem(swap.redeem_action(*beta_htlc_location)),
```
Comment (D4nte) [inline application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_erc20.rs]: 🙃 Best to be homogeneous and use `Self` everywhere

File: application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_eth.rs [hunk 2]
```diff
@@ -47,24 +47,28 @@ impl OngoingSwap<Alice<Bitcoin, Ethereum, BitcoinQuantity, EtherQuantity>> {
...
+        bitcoin::SpendOutput,
+    >;
 
-    fn actions(&self) -> Vec<AliceActionKind> {
+    fn actions(&self) -> Vec<Self::ActionKind> {
         use self::SwapStates as SS;
         match *self {
-            SS::Accepted(Accepted { ref swap, .. }) => vec![ActionKind::Fund(swap.fund_action())],
+            SS::Accepted(Accepted { ref swap, .. }) => {
+                vec![alice::ActionKind::Fund(swap.fund_action())]
```
Comment (D4nte) [inline application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_eth.rs]: 🙃 use `Self`

File: application/comit_node/src/swap_protocols/rfc003/actions/bob/btc_erc20.rs [hunk 3]
```diff
@@ -75,33 +75,31 @@ impl OngoingSwap<Bob<Bitcoin, Ethereum, BitcoinQuantity, Erc20Quantity>> {
...
+    >;
 
-    fn actions(&self) -> Vec<BobActionKind> {
+    fn actions(&self) -> Vec<Self::ActionKind> {
         use self::SwapStates as SS;
         match *self {
             SS::Start(Start { ref role, .. }) => vec![
-                ActionKind::Accept(role.accept_action()),
-                ActionKind::Decline(role.decline_action()),
+                bob::ActionKind::Accept(role.accept_action()),
```
Comment (D4nte) [inline application/comit_node/src/swap_protocols/rfc003/actions/bob/btc_erc20.rs]: 🙃 use `Self`

File: application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_erc20.rs (hunk 1 above)
Comment (luckysori) [reply application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_erc20.rs]: This is a known bug, but this PR should fix it: https://github.com/rust-lang/rust/pull/56225

File: application/comit_node/src/http_api/rfc003/with_swap_types.rs [hunk 4]
```diff
@@ -65,3 +65,67 @@ macro_rules! with_swap_types {
...
+                type Role = Bob<AL, BL, AA, BA>;
+                $fn()
+            }
+            _ => unimplemented!(),
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! with_swap_types_bob {
```
Comment (D4nte) [inline application/comit_node/src/http_api/rfc003/with_swap_types.rs]: is it a one-time use macro? is there much point?

File: application/comit_node/src/http_api/rfc003/with_swap_types.rs [hunk 5]
```diff
@@ -65,3 +65,67 @@ macro_rules! with_swap_types {
...
+macro_rules! _match_role_bob {
+    ($role:ident, $fn:tt) => {
+        #[allow(clippy::redundant_closure_call)]
+        match $role {
+            RoleKind::Bob => {
+                #[allow(dead_code)]
+                type Role = Bob<AL, BL, AA, BA>;
+                $fn()
+            }
+            _ => unimplemented!(),
```
Comment (D4nte) [inline application/comit_node/src/http_api/rfc003/with_swap_types.rs]: when do we implement that?

File: application/comit_node/src/http_api/rfc003/with_swap_types.rs (hunk 4 above)
Comment (luckysori) [reply application/comit_node/src/http_api/rfc003/with_swap_types.rs]: It saves us about 60 lines of duplicated code for every `(alpha_ledger, beta_ledger, alpha_asset, beta_asset)` tuple.

Comment (D4nte) [reply application/comit_node/src/http_api/rfc003/with_swap_types.rs]: Ok I understand know, should have said that the macro repeats the passed arguments... 😄

**Pull request merged**